txn_writer.go

```diff
@@ -113,12 +113,52 @@ func (diff *LockDiff) Format() (string, error) {
 // Fields are only populated when there is a difference, otherwise they are empty.
 // TODO(carolynvs) this should be moved to gps
 type LockedProjectDiff struct {
-	Name     gps.ProjectRoot `json:"name"`
-	Source   *StringDiff     `json:"source,omitempty"`
-	Version  *StringDiff     `json:"version,omitempty"`
-	Branch   *StringDiff     `json:"branch,omitempty"`
-	Revision *StringDiff     `json:"revision,omitempty"`
-	Packages []StringDiff    `json:"packages,omitempty"`
+	Name     gps.ProjectRoot
```

Comment:
Sorry an emergency came up and I stopped mid review. I'll start back on this today. The current toml mapper doesn't yet support allowing a nested type to use a custom marshal function, so it was manual. I started adding that this weekend and hopefully can submit that to go-toml soon and use it.